tests/datavolume_test.go

```diff
@@ -3463,6 +3463,65 @@ var _ = Describe("[vendor:cnv-qe@redhat.com][level:component]DataVolume tests",
 			}),
 		)
 	})
+
+	Describe("extra configuration options for VDDK imports", func() {
+		It("[test_id:XXXX]succeed importing VDDK data volume with extra arguments ConfigMap set", Label("VDDK"), func() {
+			vddkConfigOptions := []string{
+				"VixDiskLib.nfcAio.Session.BufSizeIn64KB=16",
+				"vixDiskLib.nfcAio.Session.BufCount=4",
+			}
+
+			vddkConfigMap := &v1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "vddk-extras",
+				},
+				Data: map[string]string{
+					common.VddkArgsKeyName: strings.Join(vddkConfigOptions, "\n"),
+				},
+			}
+
+			_, err := f.K8sClient.CoreV1().ConfigMaps(f.Namespace.Name).Create(context.TODO(), vddkConfigMap, metav1.CreateOptions{})
+			if !k8serrors.IsAlreadyExists(err) {
+				Expect(err).ToNot(HaveOccurred())
+			}
+
+			vcenterURL := fmt.Sprintf(utils.VcenterURL, f.CdiInstallNs)
+			dataVolume := createVddkDataVolume("import-pod-vddk-config-test", "100Mi", vcenterURL)
+
+			By(fmt.Sprintf("Create new DataVolume %s", dataVolume.Name))
+			controller.AddAnnotation(dataVolume, controller.AnnPodRetainAfterCompletion, "true")
+			controller.AddAnnotation(dataVolume, controller.AnnVddkExtraArgs, "vddk-extras")
+			dataVolume, err = utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, dataVolume)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Verify PVC was created")
+			pvc, err := utils.WaitForPVC(f.K8sClient, dataVolume.Namespace, dataVolume.Name)
+			Expect(err).ToNot(HaveOccurred())
+			f.ForceBindIfWaitForFirstConsumer(pvc)
+
+			By("Wait for import to be completed")
+			err = utils.WaitForDataVolumePhase(f, dataVolume.Namespace, cdiv1.Succeeded, dataVolume.Name)
+			Expect(err).ToNot(HaveOccurred(), "DataVolume not in phase succeeded in time")
+
+			By("Find importer pods after completion")
+			pvcName := dataVolume.Name
+			// When using populators, the PVC Prime name is used to build the importer pod
+			if usePopulator, _ := dvc.CheckPVCUsingPopulators(pvc); usePopulator {
+				pvcName = populators.PVCPrimeName(pvc)
+			}
+			By("Find importer pod " + pvcName)
+			importer, err := utils.FindPodByPrefixOnce(f.K8sClient, dataVolume.Namespace, common.ImporterPodName, common.CDILabelSelector)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(importer.DeletionTimestamp).To(BeNil())
+
+			logs, err := f.RunKubectlCommand("logs", "-n", dataVolume.Namespace, importer.Name)
+			Expect(err).ToNot(HaveOccurred())
+			for _, option := range vddkConfigOptions {
+				By(fmt.Sprintf("Check for configuration value %s in nbdkit logs", option))
+				Expect(strings.Contains(logs, option)).To(BeTrue())
```

Comment:
You can use the pod logs API, check this out: https://github.com/kubevirt/containerized-data-importer/pull/3184
And if you use `.Should(ContainSubstring` I am pretty sure it'll print the logs so we can properly understand why CI fails

Reply:
Done, the end of the log gets truncated though. I am going to try showing the whole log temporarily and see if it offers any insights.

Reply:
So, it passes locally?

Reply:
Yes, it passes locally. It also correctly fails if I add another ContainSubstring line, and I can see the target strings in the log output.

Reply:
Interesting, the config argument is not getting passed to nbdkit:
```
I0109 19:07:46.993477       1 nbdkit.go:311] Start nbdkit with: ['--foreground' '--readonly' '--exit-with-parent' '-U' '/tmp/nbd.sock' '--pidfile' '/tmp/nbd.pid' '--filter=retry' '--filter=cacheextents' '/opt/testing/libvddk-test-plugin.so' 'libdir=/opt/vmware-vix-disklib-distrib' 'server=vcenter.cdi-ukbxj:8989' 'user=user' 'password=+/tmp/password2397077821' 'thumbprint=testprint' 'vm=moref=vm-21' '--verbose' '-D' 'nbdkit.backend.datapath=0' '-D' 'vddk.datapath=0' '-D' 'vddk.stats=1' 'file=[teststore] testvm/testdisk.vmdk']
```

I must have a mistake further up somewhere.

Reply:
I see, I didn't get the annotations copied over correctly through the import populator stuff.  The importer prime doesn't get the annotations, so I guess the pod doesn't get the ConfigMap mounted. I can reproduce this locally now.

Reply:
I see, yeah usually the hot path is CSI storage (and thus populators). I assume you were not seeing any KUBEVIRT_STORAGE env var, and thus falling back to legacy population with the `local` storage class.

Reply:
Yes, that was it. The tests are looking better now.